### PR TITLE
Update teams.yaml for davidhauck

### PR DESCRIPTION
**Pull request by davidhauck**
Need to add myself to the subteam as well as the parent team.

Answers to questions are in this PR: https://github.com/istio/community/pull/411

### CODE REVIEW

File: org/teams.yaml [hunk 1]
```diff
@@ -87,6 +87,7 @@ teams:
             description: Maintainers of the English documentation.
             members:
               - adammil2000
+              - davidhauck
```
Comment (frankbu) [inline org/teams.yaml:90]: Also add yourself to WG - Docs Maintainers/Infra (line 105)

Comment (davidhauck) [reply org/teams.yaml:90]: Done

**Comment by frankbu** [comment]
@istio/steering-committee Can somebody please approve this?

**Review by smawson — APPROVED** [review]

**Pull request merged**